typescript/infra/scripts/core.ts

```diff
@@ -34,29 +37,56 @@ async function main() {
         : '0xa7ECcdb9Be08178f896c26b7BbD8C3D4E844d9Ba';
 
     const signer = await impersonateAccount(deployerAddress);
-    multiProvider.setSigner(argv.fork, signer);
+    multiProvider.setSigner(fork, signer);
   }
 
-  const deployer = new HyperlaneCoreInfraDeployer(
-    multiProvider,
-    config.core,
-    environment,
+  // Write agent config indexing from the latest block numbers.
+  // For non-net-new deployments, these changes will need to be
```

Comment:
It definitely sucks. It's worth noting this is not new behavior, this is how things work today (approximately). I just documented it.

Reply:
> had an idea we could put deployed block number in immutable bytecode of contracts

I really like this, would allow us to remove some agent config as well :)

Want to file a ticket?

Reply:
https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2004